src/core/scope.c

```diff
@@ -635,7 +641,9 @@ static void scope_notify_cgroup_oom_event(Unit *u, bool managed_oom) {
         else
                 log_unit_debug(u, "Process of control group was killed by the OOM killer.");
 
-        /* This will probably need to be modified when scope units get an oom-policy */
+        if (s->oom_policy == OOM_CONTINUE)
+                return;
+
```

Comment:
what about OOM_STOP?

Reply:
If the policy is OOM_STOP, then the unit will be terminated, just like before. (As I said in the issue thread, I don't think this is a good default or even a useful behavior, but unless you want to change it, this edit does not change the old behavior, so the code is fine.)

Reply:
well, what I am saying is that if OOM_STOP is set we should not enter SCOPE_STOP_SIGKILL right-away as the code will do now, but SCOPE_STOP_SIGTERM instead.

Reply:
ah, yes. I'll fix this and the group_context issue in a bit. Thanks.